pkg/querier/worker/util.go

```diff
@@ -48,7 +49,7 @@ func newExecutionContext(workerCtx context.Context, logger log.Logger) (execCtx
 			}
 
 			level.Debug(logger).Log("msg", "querier worker context has been canceled and there's no inflight query, canceling the execution context too")
-			execCancel()
+			execCancel(cancellation.NewErrorf("querier worker context cancelled"))
```

Comment:
can we include `context.Cause(workerCtx)` in the message?

Reply:
Good idea, added in db107b9.